addons/binding/org.openhab.binding.paradoxalarm/src/main/java/org/openhab/binding/paradoxalarm/internal/communication/EvoCommunicator.java

```diff
@@ -0,0 +1,326 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.paradoxalarm.internal.communication;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.openhab.binding.paradoxalarm.internal.communication.messages.EpromRequestPayload;
+import org.openhab.binding.paradoxalarm.internal.communication.messages.HeaderMessageType;
+import org.openhab.binding.paradoxalarm.internal.communication.messages.IPPacketPayload;
+import org.openhab.binding.paradoxalarm.internal.communication.messages.ParadoxIPPacket;
+import org.openhab.binding.paradoxalarm.internal.communication.messages.RamRequestPayload;
+import org.openhab.binding.paradoxalarm.internal.exceptions.ParadoxBindingException;
+import org.openhab.binding.paradoxalarm.internal.model.ZoneStateFlags;
+import org.openhab.binding.paradoxalarm.internal.util.ParadoxUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EvoCommunicator} is responsible for handling communication to Evo192 alarm system via IP150 interface.
+ *
+ * @author Konstantin_Polihronov - Initial contribution
+ */
+public class EvoCommunicator extends GenericCommunicator implements IParadoxCommunicator {
+
+    protected final Logger logger = LoggerFactory.getLogger(EvoCommunicator.class);
+
+    private MemoryMap memoryMap;
+
+    public EvoCommunicator(String ipAddress, int tcpPort, String ip150Password, String pcPassword) throws Exception {
+        super(ipAddress, tcpPort, ip150Password, pcPassword);
+        initializeMemoryMap();
+    }
+
+    @Override
+    public List<String> readPartitionLabels() {
+        List<String> result = new ArrayList<>();
+
+        try {
+            for (int i = 1; i <= 8; i++) {
+                result.add(readPartitionLabel(i));
+            }
+        } catch (Exception e) {
+            logger.debug("Unable to retrieve partition labels. Exception: {}", e);
+        }
+        return result;
+    }
+
+    public String readPartitionLabel(int partitionNo) throws Exception {
+        logger.debug("Reading partition label: " + partitionNo);
+        if (partitionNo < 1 || partitionNo > 8) {
+            throw new Exception("Invalid partition number. Valid values are 1-8.");
```

Comment:
Don't throw this generic Exception. Either create a new exception or see if an existing exception fit.